kernel/src/main/java/org/kframework/lsp/KPos.java

```diff
@@ -0,0 +1,70 @@
+package org.kframework.lsp;
+
+import org.eclipse.lsp4j.Position;
+
+import java.util.Objects;
+
+/**
+ * KPos a position in a file starting from one.
+ * Use this class when converting positions to aviod off by one errors.
+ */
+public class KPos {
+
+    /**
+     * Line position in a document (one-based).
+     */
+    private int line;
+
+    /**
+     * Character offset on a line in a document (one-based).
+     */
+    private int character;
+
+    public KPos() {
+        line = 0;
+        character = 0;
+    }
+
+    public KPos(int line, int character) {
+        this.line = line;
+        this.character = character;
+    }
+
+    public KPos(Position pos) {
+        this.line = pos.getLine() + 1;
+        this.character = pos.getCharacter();
+    }
+
+    public Position toPosition() {
+        return new Position(line - 1, character - 1);
+    }
+
+    public int getLine() {
+        return line;
+    }
+
+    public void setLine(int line) {
+        this.line = line;
+    }
+
+    public int getCharacter() {
+        return character;
+    }
+
+    public void setCharacter(int character) {
+        this.character = character;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        KPos kPos = (KPos) o;
+        return line == kPos.line && character == kPos.character;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(line, character);
+    }
+}
```

Comment:
I'm not convinced by this class; couldn't this be implemented as a method `humanReadable` on `Position` that gives you back the 1-based position from the 0-based internal representation?

Reply:
The `Position` class is part of the lsp library. So I can't modify it.
I made a new `KPos` class in order to make sure I always get the correct arguments to the klsp.
Basically, I'm relying on the type checker to make sure I always do the correct translations.

Reply:
That makes sense; perfectly fine if Position isn't our code.